### DISCUSSION TITLE
Display app list on client-store

### DISCUSSION THREAD

**Discussion by jonas-martinez** [lenra-io/lenra_cli#56] [General]
# Display app list on client-store

We should add a new page on the `client-store` project to display all of the applications that the user has access to.

## Which app to display ?

Applications that have to be displayed to the user are :
- The ones that the user created 
- The ones that the user has been invited to
- The ones that are public ? 

But what if there is thousands of public apps, we do not want to show them to the user directly on his main page.

**Here is a solution :**
Show the apps that the user created. Show the ones he has been invited to only if he has opened it at least once. Show the ones that are public only if he has opened it at least once.

We then must add "tabs" to display more "invited to" and "public" apps.
We could have 3 tabs :
- My apps 
- Invited to
- Store (public apps)

The user has the choice to remove an app from his "My apps" tab. (Maybe this feature needs a new discussion)

### For the app list part.
**Cards** that are made of :
- Icon
- App name
- Buttons (Open, More informations, ...)

### For the app page part

Display everything the user needs to know on this application. (app name, description, permissions, etc ...)

What is your opinion on this @lenra-io/devs ?

**Comment by shiipou** [comment]
In think we can call the `invited to` -> `Installed`, `owned` or `library` (my favorite) So any app from the store can be added to our account must be listed here among invited apps.

So for me, any app where `env` is public must be shown here.

Other points of this discussion seems good to me.

**Reply by jonas-martinez** [reply]
So with this change we would have these 3 tabs :
- My apps
- Library
- Store (public apps)

Is that right @shiipou ?

**Reply by shiipou** [reply]
> So with this change we would have these 3 tabs :
> - My apps
> - Library
> - Store (public apps)
> 
> Is that right @shiipou ?

Yes That seems better for me.

**Comment by pichoemr** [comment]
What do you mean by "app page"?

For me, we need two pages to display the apps, we can have a client-store homepage that displays the latest open/favorite/created apps.
and a store page with all public applications

**Reply by jonas-martinez** [reply]
By app page I mean the page that shows the information about one specific application. 

For example (using the Play Store), this would be the store page
![image](https://user-images.githubusercontent.com/36544012/195054649-b0754c8f-b29b-48d4-a619-19df432f1e2c.png)

And this would be the app page
![image](https://user-images.githubusercontent.com/36544012/195054758-1bd62e60-35c4-4d23-b61f-f9d40ad9e2c0.png)

**Comment by jonas-martinez** [comment]
## EDIT

I commented on the wrong discussion, this text below belongs to https://github.com/lenra-io/Lenra/discussions/82

> # The work that needs to be done
> 
> Currently, we have a variable used at multiple places in the backoffice code ( `selected_app` ).
> This variable is stored inside of the `UserApplicationModel` which can be called from anywhere in the backoffice.
> This allows us to create a new page that will display the developer's application list. He will have to choose the app he wants to use for his session. Set the `selected_app` value to match this choice and we're done for this feature.
> 
> **Recap :** 
> - Create a new page to show the developer's application list if the `selected_app` variable is not set.
>   - Show list of apps
>   - Add a button to create a new app (will open the `CreateProjectForm`)
> - Add a button of the OverviewPage to redirect to the page created in the previous task.
> - That's all ??

**Comment by taorepoara** [comment]
be careful to do not over-scope this discussion. The public app list is not part of this discussion seens is the app store and it's not for now. The app invitation is also out of this scope and should be the subject of another discussion.

I think a home page with the list of the apps having an account for the user is the simplest. For the app ordering we can start with alphabetical or last use (I prefer alphabetical).

**Reply by jonas-martinez** [reply]
Oops I commented in the wrong discussion, that's why it is out of scope ^^

**Comment by taorepoara** [comment]
Do we manage the app logo ?

What do we do for apps that does not have a logo ? It can't be published ? Do we generate a custom one with the first letter of the app like the Google accounts ?

**Reply by jonas-martinez** [reply]
Yes we can manage the app logo but it can be a future feature as we can generate logos with the first letter of the app for the moment.

**Comment by Nesqwik** [comment]
Instead of tabs for each list, we can have all three list in the same page.
Each list scroll horizontally, the store page can scroll vertically.
This allow us to add new list if needed (popular, games, work and so on..)

If you open the play store on android, it works a bit like that.
In the beginning we will not have a lot of apps, this could work just fine :) 

Later on, we will have to add a full-text search system, multiple tabs with multiple sub-tab and so on..
But for now, keeping it simple using a simple page could be better ?

Anyway a designer should work on this one to create a much better UI

**Reply by taorepoara** [reply]
As said earlier, this discussion is only for the home page, to let the user find his apps, not for the store
